<launch>
  <arg name="manager" default="zvision_lidar_nodelet_manager" />
  <arg name="model" default="ML30SA1" />
  <arg name="device_ip" default="192.168.1.200" />
  <arg name="udp_port" default="2368" />
  <arg name="angle_path" default="$(find zvision_lidar_pointcloud)/data/ML30S_A1_Default.cal" />
  <arg name="pcap" default="$(find zvision_lidar_pointcloud)/data/ML30S_A1_Default.pcap" />
  <arg name="use_blue_gold_color_scheme" default="false" />

  <!-- load driver nodelet into it (udp packets)-->
  <node pkg="zvision_lidar_driver" type="zvision_lidar_node" name="zvision_lidar_node_driver"
      output="screen">
    <param name="model" value="$(arg model)"/>
    <param name="device_ip" value="$(arg device_ip)" />
    <param name="udp_port" value="$(arg udp_port)" />
	  <param name="pcap" value="$(arg pcap)" />
  </node>

  <!-- load driver nodelet into it (udp packet to pointcloud)-->
  <node pkg="zvision_lidar_pointcloud" type="zvision_lidar_cloud_node" name="zvision_lidar_cloud_node_cloud"
      output="screen">
    <param name="model" value="$(arg model)"/>
    <param name="angle_path" value="$(arg angle_path)" />
    <param name="device_ip" value="$(arg device_ip)" />
    <param name="x_tra" value="0.0" />
    <param name="y_tra" value="0.0" />
    <param name="z_tra" value="0.0" />
    <param name="x_rot" value="0.0" />
    <param name="y_rot" value="0.0" />
    <param name="z_rot" value="0.0" />
    <param name="use_lidar_time" value="false" />
    <param name="use_outlier_removal" value="false"/>
    <param name="outlier_th" value="0.05"/>
    <param name="use_blue_gold_color_scheme" value="$(arg use_blue_gold_color_scheme)"/>
  </node>
  
  
  <node if="$(arg use_blue_gold_color_scheme)" name="rviz" pkg="rviz" type="rviz"  args="-d $(find zvision_lidar_pointcloud)/rviz_cfg/zvision_lidar_colored.rviz" />
  <node unless="$(arg use_blue_gold_color_scheme)" name="rviz" pkg="rviz" type="rviz"  args="-d $(find zvision_lidar_pointcloud)/rviz_cfg/zvision_lidar.rviz" />

</launch>
